lib/rules/no-constant-binary-expression.js

```diff
@@ -45,6 +45,12 @@ function hasConstantNullishness(scope, node) {
             return (functionName === "Boolean" || functionName === "String" || functionName === "Number") &&
                 isReferenceToGlobalVariable(scope, node.callee);
         }
+        case "LogicalExpression": {
+            const isLeftConstant = hasConstantNullishness(scope, node.left);
+            const isRightConstant = hasConstantNullishness(scope, node.right);
+
+            return isLeftConstant || isRightConstant;
+        }
```

Comment:
These would be false positives, as the left-hand sides of these `??` do not have constant nullishness so the expressions neither always nor never short-circuit:

```js
const first = (foo && true) ?? bar;
//             ^^^^^^^^^^^ Unexpected constant nullishness on the left-hand side of a `??` expression.

const second = null ?? foo ?? bar;
//             ^^^^^^^^^^^ Unexpected constant nullishness on the left-hand side of a `??` expression.

const third = foo ?? null ?? bar;
//            ^^^^^^^^^^^ Unexpected constant nullishness on the left-hand side of a `??` expression.

```

We should check if the operator is `??` and if its right side is non-nullish.

Reply:
Thank you for review! I tried to fix in 7c91ba8.